CONTRIBUTING.md

```diff
@@ -0,0 +1,119 @@
+# Contributing to GBIF Pipelines
+
+The GBIF Pipelines development community welcomes contributions from anyone! Thank you.
+
+If you have questions please open [an issue](https://github.com/gbif/pipelines/issues/new) or contact the developers at informatics@gbif.org.
+
+There are different ways you can contribute that help this project:
+- Log [issues](https://github.com/gbif/pipelines/issues) and help document a bug or specify new functionality
+- Improve documentation
+- Provide code submissions that address issues or bring new functionalities
+- Help test the functionalities offered in the project
+- Help improve this guide
+
+## Contributing code
+
+Below is a tutorial for contributing code to Pipelines, covering our tools and typical process in detail.
+
+### Prerequisites
+
+To contribute code, you need
+
+- a GitHub account
+- a Linux, MacOS, or Microsoft Windows development environment with Java JDK 8 installed
+- Maven (version 3.6+)
+
+### Connect with the Pipeline community and share your intent
+
+- This is a very active project, with dependencies coming from those running in production.
+- It is always worthwhile announcing your intention, so the community can offer guidance.
+- Please always start with an issue to capture discussion
+
+### Development setup
+
+#### Command line
+
+This project uses [Apache Maven](https://maven.apache.org/run.html) to build. The following should be enough to checkout and build the project.
+
+```
+git clone git@github.com:gbif/pipelines.git
+cd pipelines
+mvn clean package
+```
+To skip integration tests, run `mvn clean package -DskipITs`.
+
+We use [Google Java Format](https://plugins.jetbrains.com/plugin/8527-google-java-format) for code styling.
+
+From the command line you can check the style using `mvn spotless:check` and fixup styling issues using `mvn spotless:apply`.
+
+#### IDE Setup
+
+We recommend the following toolset, and are able to answer questions on this configuration:
+
+- Use [IntelliJ IDEA Community](https://www.jetbrains.com/idea/download/) (or better),
+- Use the [Google Java Format](https://plugins.jetbrains.com/plugin/8527-google-java-format). Please do not reformat existing code, only changes and new code.
+- The project uses [Project Lombok](https://projectlombok.org/). Please install the [Lombok plugin for IntelliJ IDEA](https://plugins.jetbrains.com/plugin/6317-lombok-plugin).
+- Because the project uses [Error Prone](https://code.google.com/p/error-prone) you may have issues during the build process from IDEA.  To avoid these issues please install the [Error Prone compiler integration plugin](https://plugins.jetbrains.com/plugin/7349-error-prone-compiler-integration) and build the project using the [`error-prone java compiler`](https://code.google.com/p/error-prone) to catch common Java mistakes at compile-time. To use the compiler, go to _File_ → _Settings_ → _Compiler_ → _Java Compiler_ and select `Javac with error-prone` in the `Use compiler` box.
```

Comment:
Theres ongoing issues with error prone and Intellij (see slack discussion with Vicente - i have  had the same issues)

Reply:
Thanks. I suggest we also get mavenized `errorprone` options working too for the reason it is more robust than relying on IDE use. I haven't explored that yet